src/System.Threading.Thread/src/System/Threading/Thread.cs

```diff
@@ -0,0 +1,420 @@
+        private void ThreadMain_ThreadStart()
+        {
+            t_currentThread = this;
+
+            var start = m_start;
+            m_start = null;
+            Debug.Assert(start is ThreadStart);
+            ((ThreadStart)start)();
+        }
+
+        private void ThreadMain_ParameterizedThreadStart(object parameter)
+        {
+            t_currentThread = this;
+
+            var start = m_start;
+            m_start = null;
+            Debug.Assert(start is ParameterizedThreadStart);
+            ((ParameterizedThreadStart)start)(parameter);
+        }
+
+        public static Thread CurrentThread
+        {
+            get
+            {
+                var currentThread = t_currentThread;
+                if (currentThread == null)
+                {
+                    t_currentThread = currentThread = new Thread(RuntimeThread.CurrentThread);
+                }
+                return currentThread;
+            }
+        }
+
+        private void RequireCurrentThread()
+        {
+            if (this != CurrentThread)
+            {
+                throw new InvalidOperationException(SR.Thread_Operation_RequiresCurrentThread);
+            }
+        }
+
+        public CultureInfo CurrentCulture
+        {
+            get
+            {
+                RequireCurrentThread();
+                return CultureInfo.CurrentCulture;
+            }
+            set
+            {
+                RequireCurrentThread();
+                CultureInfo.CurrentCulture = value;
+            }
+        }
+
+        public CultureInfo CurrentUICulture
+        {
+            get
+            {
+                RequireCurrentThread();
+                return CultureInfo.CurrentUICulture;
+            }
+            set
+            {
+                RequireCurrentThread();
+                CultureInfo.CurrentUICulture = value;
+            }
+        }
+
+        public static IPrincipal CurrentPrincipal
+        {
+            get
+            {
+                return CurrentThread.m_principal;
+            }
+            set
+            {
+                CurrentThread.m_principal = value;
+            }
+        }
+
+        public ExecutionContext ExecutionContext => ExecutionContext.Capture();
+        public bool IsAlive => m_runtimeThread.IsAlive;
+        public bool IsBackground { get { return m_runtimeThread.IsBackground; } set { m_runtimeThread.IsBackground = value; } }
+        public bool IsThreadPoolThread => m_runtimeThread.IsThreadPoolThread;
+        public int ManagedThreadId => m_runtimeThread.ManagedThreadId;
+        public string Name { get { return m_runtimeThread.Name; } set { m_runtimeThread.Name = value; } }
+        public ThreadPriority Priority { get { return m_runtimeThread.Priority; } set { m_runtimeThread.Priority = value; } }
+        public ThreadState ThreadState => m_runtimeThread.ThreadState;
+
+        public void Abort()
+        {
+            throw new PlatformNotSupportedException();
```

Comment:
I do not think so. It would not match what it does on desktop - ThreadAbortException thrown by Abort does not behave as regular exception.